HopsanGenerator/templates/fmu_hopsan.cpp

```diff
@@ -138,45 +149,88 @@ void hopsan_get_message(hopsan_message_callback_t message_callback, void* userSt
 
 double hopsan_get_real(int vr)
 {
-    return (*dataPtrs[vr]);
+    if(vr < sizeof(dataPtrs)/sizeof(double*)) {
+        return (*dataPtrs[vr]);
+    }
+    if(realParametersMap.count(vr)) {
```

Comment:
Maybe it is better to add a static const variable with the max vr value, to avoid counting all of the time.
But maybe not such a  big deal since it only happens for parameters that are rarely accessed.

Reply:
It was even easier to just use a hard-coded value, since the code is auto generated anyway. It is actually a big deal, since this check is performed for every getReal/setReal call, also for input/output variables.

Reply:
I meant the parametermap.count(), could add an other numparameters, but not so important.

Reply:
Now I see what you mean. But parameters are only set at the beginning of the simulation anyway, so it should not be performance critical. 

I was first going to use `contains()`, but I realized that it is not compatible with older compilers.